debezium-server/debezium-server-http/src/main/java/io/debezium/server/http/HttpChangeConsumer.java

```diff
@@ -99,27 +110,42 @@ public void handleBatch(List<ChangeEvent<Object, Object>> records, DebeziumEngin
             LOGGER.trace("Received event '{}'", record);
 
             if (record.value() != null) {
-                String value = (String) record.value();
-                HttpResponse r;
-
-                HttpRequest request = requestBuilder.POST(HttpRequest.BodyPublishers.ofString(value)).build();
-
-                try {
-                    r = client.send(request, HttpResponse.BodyHandlers.ofString());
-                }
-                catch (IOException ioe) {
-                    throw new InterruptedException(ioe.toString());
-                }
-
-                if ((r.statusCode() == HTTP_OK) || (r.statusCode() == HTTP_NO_CONTENT) || (r.statusCode() == HTTP_ACCEPTED)) {
-                    committer.markProcessed(record);
-                }
-                else {
-                    LOGGER.info("Failed to publish event: " + r.body());
+                int attempts = 0;
+                while (!recordSent(record)) {
+                    attempts++;
+                    if (attempts >= retries) {
+                        throw new DebeziumException("Exceeded maximum number of attempts to publish event " + record);
+                    }
+                    Thread.sleep(retryInterval);
```

Comment:
Would it make sense to use `Metronome` here rather than the explicit `Thread#sleep` method call?

Reply:
Yes, definitely makes sense. I vaguely remembered Debezium has some utility for it and wanted to use it, but failed to remember/find it. Thanks!